azure/services/async/async.go

```diff
@@ -29,6 +29,22 @@ import (
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
+// Service is an implementation of the AsyncReconciler interface. It handles asynchronous creation and deletion of resources.
```

Comment:
```suggestion
// Service is an implementation of the Reconciler interface. It handles asynchronous creation and deletion of resources.
```

since it was renamed

Reply:
looks like the rename was reverted